core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -168,7 +313,90 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     }
 
     @Override
+    public boolean doEquals(GeoBoundingBoxQueryBuilder other) {
+        return Objects.equals(queryName, other.queryName) &&
+                Objects.equals(topLeft, other.topLeft) &&
+                Objects.equals(bottomRight, other.bottomRight) &&
+                Objects.equals(type, other.type) &&
+                Objects.equals(fieldName, other.fieldName);
+    }
+
+    @Override
+    public int doHashCode() {
+        return Objects.hash(topLeft, bottomRight, type, fieldName);
+    }
+
+    @Override
+    public GeoBoundingBoxQueryBuilder doReadFrom(StreamInput in) throws IOException {
+        String fieldName = in.readString();
+        GeoBoundingBoxQueryBuilder geo = new GeoBoundingBoxQueryBuilder(fieldName);
+        double lat = in.readDouble();
+        double lon = in.readDouble();
+        geo.topLeft = new GeoPoint(lat, lon);
+        lat = in.readDouble();
+        lon = in.readDouble();
+        geo.bottomRight = new GeoPoint(lat, lon);
+        geo.type = Type.readTypeFrom(in);
+        return geo;
+    }
+
+    @Override
+    public void doWriteTo(StreamOutput out) throws IOException {
+        out.writeString(fieldName);
+        out.writeDouble(topLeft.getLat());
+        out.writeDouble(topLeft.getLon());
+        out.writeDouble(bottomRight.getLat());
+        out.writeDouble(bottomRight.getLon());
+        type.writeTo(out);
+    }
+
+    @Override
     public String getName() {
         return NAME;
     }
+
+    public enum Type implements Writeable<Type> {
+        MEMORY(0), INDEXED(1);
+
+        private final int ordinal;
+
+        private static final Type PROTOTYPE = MEMORY;
+
+        private Type(int ordinal) {
+            this.ordinal = ordinal;
+        }
+
+        @Override
+        public Type readFrom(StreamInput in) throws IOException {
+            int ord = in.readVInt();
+            for (Type type : Type.values()) {
```

Comment:
Since there are only two allowed values, I was thinking about whether the loop makes sense here of if simple if or switch would be more efficient?